File: src/main/java/ca/cutterslade/experiment/gdk/GreeterModule.java
package ca.cutterslade.experiment.gdk;

import java.util.Collections;
import java.util.List;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.grpc.server.ForGrpcService;
import io.grpc.ServerInterceptor;

@Module
public abstract class GreeterModule {
  @Binds
  public abstract GreeterServiceDefinition greeterComponent(GreeterComponent component);

  @Provides
  @ForGrpcService(GreeterGrpc.class)
  public static List<? extends ServerInterceptor> interceptors() {
    return Collections.emptyList();
  }
}

File: src/main/kotlin/ca/cutterslade/experiment/gdk/GreeterModule.kt
package ca.cutterslade.experiment.gdk

import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.grpc.server.ForGrpcService
import io.grpc.ServerInterceptor

@Module
abstract class GreeterModule {
  @Binds
  abstract fun greeterComponent(component: GreeterComponent): GreeterServiceDefinition

  companion object {
    @Provides
    @ForGrpcService(GreeterGrpc::class)
    @JvmStatic
    fun interceptors(): List<ServerInterceptor> = listOf()
  }
}
